### PR TITLE
Fix: Expect return true immediately if If-None-Match matches the ETag header (#35)

### PR DESCRIPTION
**Pull request by bigbigDreamer**
### Main Changes
Expect  return true immediately if If-None-Match matches the ETag header.

### Context
Closes https://github.com/jshttp/fresh/issues/35

### CODE REVIEW

**Review by UlisesGascon — APPROVED** [review]
Thanks for making this work @bigbigDreamer!

AFAIK, seems like this is a **major change in terms of semver**. I will love to have some additional reviews before we merge it (I am not an expert on `fresh`) :+1:

**Comment by ctcpip** [comment]
I don't have the requisite permission to approve, but 👍 ✅

**Comment by ctcpip** [comment]
> AFAIK, seems like this is a **major change in terms of semver**.

there is no perfect answer, but you could argue that a breaking change caused by spec-compliance where you don't do a major version bump does not violate semver.  unrelated but similar issue I ran into last week:  `github.com/bcgit/bc-java/issues/1641`

**Review by rxmarbles — COMMENTED** [review]
Also don't have perms to approve but a ➕1 from me 😄

**Review by ctcpip — COMMENTED** [review]
I don't have the requisite permission to approve, but 👍 ✅

(previous comment doesn't show up in reviews)

**Review by import-brain — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -49,26 +49,27 @@ function fresh (reqHeaders, resHeaders) {
...
+      return true
+    }
     var etag = resHeaders.etag
 
     if (!etag) {
       return false
     }
 
-    var etagStale = true
+    var etagStale = false
```
Comment (marco2216) [inline index.js:62]: Does this name make sense now? Shouldn't it be changed to `etagFresh`?

Comment (bigbigDreamer) [reply index.js]: Agreed, it has been modified.

**Comment by wesleytodd** [comment]
I am adding this to the list of required releases before we land `express@5`. I have not had a chance to read all the relevant context, but it does seem that we could land this as a major along with dropping support for some old node versions to release along side v5.

**Review by ctcpip — APPROVED** [review]

**Review by jonchurch — APPROVED** [review]

**Comment by jonchurch** [comment]
Updated the PR to early return from the for loop. The change to the flow of the code allows us to do that, while previously we fell through to give the other header a chance to invalidate.

When we have an if-none-match, we must decide fresh based on that

**Pull request merged**